include/acutest.h

```diff
@@ -384,12 +385,7 @@ static jmp_buf test_abort_jmp_buf__;
     test_timer_init__(void)
     {
         if(test_timer__ == 1)
-    #ifdef CLOCK_MONOTONIC_RAW
```

Comment:
Could you please explain this more? 

My motivation was to provide an elapsed time as with a stopwatch, without some calendar events like e.g. leap seconds, switches to summer time zones, and especially NTP syncing to interfere, if the system is capable of that.

But I admit I am no expert at Linux timer so maybe I did the direct opposite?

Reply:
Historically, CLOCK_MONOTONIC_RAW was provided to give a faster way to get fine grained performance numbers -- there is math done for adjtime() -- and also if you wanted to have timing data that was subject to modifications by NTP.

The thing is, NTP is actually supposed to be making those adjustments for a reason -- and that is that processor clocks drift (they're pretty subject to changes due to thermal conditions for example).

Generally this mostly means that if you want to count cycles, CLOCK_MONOTONIC_RAW might be what you want.  But most of the time, you want real-world time, which is what CLOCK_MONOTONIC is for, although there is also CLOCK_REALTIME, but the problem with CLOCK_REALTIME is that it is subject to major clock changes e.g. due to timezone, etc. and can even go backwards.

CLOCK_MONOTONIC should mostly give the same amount of real-world time for each quanta *because* it is adjusting for differences in the system clock by synchronizing with external timing sources (assuming you have a working NTP implementation).

Basically, my recommendation is never to use CLOCK_MONOTONIC_RAW unless you know you have a specific requirement to avoid the minor slewing that NTP makes via adjtime().  The only times that requirement really comes up is if you're implementing another clock synchronization system (a different ntpd for example) or you want to count *cycles* irrespective of vagaries in the system clock frequency.

Mostly this means, avoid CLOCK_MONOTONIC_RAW for ~everyone.

Reply:
Btw, CLOCK_MONOTONIC will not be affected by timezones, leap seconds, or major events like ntpdate running to reset the clock.  It will only be affected by *very* small frequency adjustments made by NTP to accommodate clock drift.  (I forget the limit it will make, but the frequency change is a small percentage.)

Reply:
> The thing is, NTP is actually supposed to be making those adjustments for a reason -- and that is that processor clocks drift (they're pretty subject to changes due to thermal conditions for example).

I understand that.

My thinking was going as this: I assume the drift may accumulate for a long time period to reach some substantial gap from `CLOCK_MONOTONIC`. But then, when it does, the adjustment can occur exactly in the very short time period while the test is being executed. The result would then be likely more off then without the adjustment. And I guess the vast majority of tests is likely executed in < 1 sec time frame.

Is my assumption wrong and the NTP is synced so often this plays no real role?

Reply:
Oops. Race condition. Your latest comment seen after page refresh explains that.

Reply:
Yes, you're misunderstanding how this works.  The concerns you have would apply to other clocks, but not CLOCK_MONOTONIC.

NTP via adjtime adjusts only the clock *frequency*, and only by small amounts.  This has a few ramifications:

1. The monotonic nature (always increasing a single microsecond at a time) is preserved.
2. You will never see a large difference in the real-world time between one sampling of CLOCK_MONOTONIC and the next.